### PR TITLE
Bump go version to 1.16

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR bumps the go version to 1.16 in preparation to use the next architect version in architect-orb (https://github.com/giantswarm/architect-orb/pull/235).

Tested in 
- https://github.com/giantswarm/devctl/pull/225 Failing
- https://github.com/giantswarm/apptest/pull/81 Works
- https://github.com/giantswarm/apptestctl/pull/104 Failing

Changes:

- updates the used golang version to 1.16
- ~Updates the generated files from `devctl` (In preparation towards adding this repository to the automation)~
- Adds a replacement for github.com/coreos/etcd

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,22 +1,30 @@
 # DO NOT EDIT. Generated with:
 #
-#    devctl gen dependabot
+#    devctl@4.3.0
```
Comment (kopiczko) [inline .github/dependabot.yml:3]: Doesn't giantswarm/github work for architect?

Comment (ubergesundheit) [reply .github/dependabot.yml]: Thanks for doing this for me :)

**Review by kopiczko — APPROVED** [review]

**Comment by ubergesundheit** [comment]
@kopiczko what do you think about the failing ci runs linked above? I think this PR needs to be on hold until we figure out if the failing builds are caused by changes in this PR or in the projects

**Comment by MarcelMue** [comment]
Very weird as the failure points to missing imports in go.sum - I could find nothing online about this at a glance.

**Comment by ubergesundheit** [comment]
I guess reading release logs is required for this one :)

> https://golang.org/doc/go1.16
> use of go get to build and install packages is deprecated

`goimports` is installed using `go get`. I've changed this in `architect-orb`, now it just fails on the nancy step :)

**Pull request merged**